src/SUMMARY.md

```diff
@@ -26,19 +26,19 @@
   - [References](basic-syntax/references.md)
     - [Dangling References](basic-syntax/references-dangling.md)
   - [Slices](basic-syntax/slices.md)
-    - [`String` vs `str`](basic-syntax/string-slices.md)
+    - [String vs str](basic-syntax/string-slices.md)
```

Comment:
Why have you removed formatting from this file? It doesn't seem to be related to adding support for translations.

Reply:
Good question! :smile: It turns out that mdbook is removing the formatting from these titles: [summary.rs](https://github.com/rust-lang/mdBook/blob/a8c37ceacee614171a59bf5ca969598b25089d44/src/book/summary.rs#L577).

The result of this stripping ends up as [`Chapter::name`](https://github.com/rust-lang/mdBook/blob/8cdb8d03672f3b5a4a7c6ecf3942fd74c029562b/src/book/book.rs#L157). We use this field when we extract the strings to translate: so instead of saving `"foo _bar_ baz"`, we would end up saving `"foo bar baz"` in `messages.pot`.

I search the `SUMMARY.md` file for the strings we translate so that I can add a proper filename and _line number_ to each translated string. When the string to translate has been altered compared to what is on disk, I cannot find it any longer.

I could potentially redo the parsing of mdbook and strip the formatting while tracking the line numbers. However, since the formatting doesn't show up in the book, it seemed fine to strip it from the source directly.

Reply:
Okay, fair enough. Maybe it would be a good idea to leave a comment in this file saying not to use formatting for chapter names (and explaining why).

Reply:
I didn't realize this yesterday, but the problem will actually be caught by the test I added: when you run `mdbook-xgettext`, it will error out if it cannot map the chapter names back to lines in `SUMMARY.md`. So we'll immediately know if someone accidentally tries to add such a title to our material.

I've updated the error message a little in that case. If/when we publish `mdbook-xgettext` and `mdbook-gettext`  to crates.io for others to use, we can consider making this less strict.